### PR TITLE
Do complete Refactoring of the UI

### PR DESCRIPTION
**Pull request by RaHoni**
This includes 
- splitting the commandhandler in destinctive functions for easier return value handling
- changing the ipc socket to send serialized class instances
- adding a cli to the console

### CODE REVIEW

File: src/main/scala/PrivateHome/UI/repl.scala [hunk 1]
```diff
@@ -25,52 +25,64 @@ import org.slf4j.LoggerFactory
 import java.io._
 
 
-
 object repl {
-  private val logger = LoggerFactory.getLogger(this.getClass)
   val socketPath = "/tmp/privatehome2.sock"
```
Comment (maximiliani) [inline src/main/scala/PrivateHome/UI/repl.scala:29]: Why is the socket named "privatehome2.sock"?

File: src/main/scala/PrivateHome/data.scala [hunk 2]
```diff
@@ -223,7 +225,7 @@ object data {
       select.from(device as m)
     }.map(rs => device(rs)).list().apply().foreach(d => {
       d.switchtype match {
-        case "mqtt" =>
+        case Mqtt =>
```
Comment (maximiliani) [inline src/main/scala/PrivateHome/data.scala:228]: All uppercase would look nicer, as it is an own type, right?

**Review by maximiliani — DISMISSED** [review]
Good job.  For the next time please write some more comments and do some more testing.

File: src/main/scala/PrivateHome/data.scala (hunk 2 above)
Comment (RaHoni) [reply src/main/scala/PrivateHome/data.scala:228]: Yea

File: src/main/scala/PrivateHome/UI/repl.scala (hunk 1 above)
Comment (RaHoni) [reply src/main/scala/PrivateHome/UI/repl.scala:29]: Because of the old socketfile which was simply privatehome.sock

**Comment by RaHoni** [comment]
This isn't ready yet because there are missing test and much more

File: src/main/scala/PrivateHome/UI/repl.scala (hunk 1 above)
Comment (RaHoni) [reply src/main/scala/PrivateHome/UI/repl.scala]: This is fixed now

**Review by maximiliani — CHANGES_REQUESTED** [review]

**Review by maximiliani — APPROVED** [review]

**Comment by RaHoni** [comment]
This is far enough to be merged and extended in the future

**Pull request merged**